File: test/aidbox_sdk/snapshots/dotnethl7-fhir-r4-core/DeviceDefinition.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class DeviceDefinition : DomainResource, IResource
{
    public DeviceDefinition_DeviceName[]? DeviceName { get; set; }
    public Base.ProductShelfLife[]? ShelfLifeStorage { get; set; }
    public DeviceDefinition_Property[]? Property { get; set; }
    public string? ManufacturerString { get; set; }
    public string? ModelNumber { get; set; }
    public DeviceDefinition_UdiDeviceIdentifier[]? UdiDeviceIdentifier { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public Base.ResourceReference? ManufacturerReference { get; set; }
    public DeviceDefinition_Capability[]? Capability { get; set; }
    public DeviceDefinition_Specialization[]? Specialization { get; set; }
    public Base.ResourceReference? ParentDevice { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.CodeableConcept[]? LanguageCode { get; set; }
    public Base.CodeableConcept[]? Safety { get; set; }
    public DeviceDefinition_Material[]? Material { get; set; }
    public string? Url { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public object? Manufacturer    
    {
        get
        {
            if (ManufacturerString is not null)
            {
                return ManufacturerString;
            }
    
            if (ManufacturerReference is not null)
            {
                return ManufacturerReference;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(string))
            {
                ManufacturerString = (string)value;
                return;
            }
    
            if (value?.GetType() == typeof(Base.ResourceReference))
            {
                ManufacturerReference = (Base.ResourceReference)value;
                return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.Quantity? Quantity { get; set; }
    public string[]? Version { get; set; }
    public Base.ContactPoint[]? Contact { get; set; }
    public Base.ResourceReference? Owner { get; set; }
    public string? OnlineInformation { get; set; }
    public Base.ProdCharacteristic? PhysicalCharacteristics { get; set; }

    public class DeviceDefinition_DeviceName : BackboneElement
    {
        public required string Name { get; set; }
        public required string Type { get; set; }
    }

    public class DeviceDefinition_Property : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public Base.CodeableConcept[]? ValueCode { get; set; }
        public Base.Quantity[]? ValueQuantity { get; set; }
    }

    public class DeviceDefinition_UdiDeviceIdentifier : BackboneElement
    {
        public required string Issuer { get; set; }
        public required string Jurisdiction { get; set; }
        public required string DeviceIdentifier { get; set; }
    }

    public class DeviceDefinition_Capability : BackboneElement
    {
        public required Base.CodeableConcept Type { get; set; }
        public Base.CodeableConcept[]? Description { get; set; }
    }

    public class DeviceDefinition_Specialization : BackboneElement
    {
        public string? Version { get; set; }
        public required string SystemType { get; set; }
    }

    public class DeviceDefinition_Material : BackboneElement
    {
        public bool? Alternate { get; set; }
        public required Base.CodeableConcept Substance { get; set; }
        public bool? AllergenicIndicator { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/DeviceDefinition.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import CodeableConcept
from base import ContactPoint
from base import ProdCharacteristic
from base import Quantity
from base import DomainResource
from base import ProductShelfLife
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class DeviceDefinitionDeviceName(BackboneElement):
    name: str
    type: str

@dataclass(kw_only=True)
class DeviceDefinitionProperty(BackboneElement):
    type: CodeableConcept
    value_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    value_quantity: Optional[List[Quantity]] = field(default_factory=list)

@dataclass(kw_only=True)
class DeviceDefinitionUdiDeviceIdentifier(BackboneElement):
    issuer: str
    jurisdiction: str
    device_identifier: str

@dataclass(kw_only=True)
class DeviceDefinitionCapability(BackboneElement):
    type: CodeableConcept
    description: Optional[List[CodeableConcept]] = field(default_factory=list)

@dataclass(kw_only=True)
class DeviceDefinitionSpecialization(BackboneElement):
    version: Optional[str] = None
    system_type: str

@dataclass(kw_only=True)
class DeviceDefinitionMaterial(BackboneElement):
    alternate: Optional[bool] = None
    substance: CodeableConcept
    allergenic_indicator: Optional[bool] = None

@dataclass(kw_only=True)
class DeviceDefinition(DomainResource):
    device_name: Optional[List[DeviceDefinitionDeviceName]] = field(default_factory=list)
    shelf_life_storage: Optional[List[ProductShelfLife]] = field(default_factory=list)
    property: Optional[List[DeviceDefinitionProperty]] = field(default_factory=list)
    manufacturer_string: Optional[str] = None
    model_number: Optional[str] = None
    udi_device_identifier: Optional[List[DeviceDefinitionUdiDeviceIdentifier]] = field(default_factory=list)
    type: Optional[CodeableConcept] = None
    manufacturer_reference: Optional[Reference] = None
    capability: Optional[List[DeviceDefinitionCapability]] = field(default_factory=list)
    specialization: Optional[List[DeviceDefinitionSpecialization]] = field(default_factory=list)
    parent_device: Optional[Reference] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    language_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    safety: Optional[List[CodeableConcept]] = field(default_factory=list)
    material: Optional[List[DeviceDefinitionMaterial]] = field(default_factory=list)
    url: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    quantity: Optional[Quantity] = None
    version: Optional[List[str]] = field(default_factory=list)
    contact: Optional[List[ContactPoint]] = field(default_factory=list)
    owner: Optional[Reference] = None
    online_information: Optional[str] = None
    physical_characteristics: Optional[ProdCharacteristic] = None